### ISSUE TITLE
Awesome Machine Learning

### ISSUE THREAD

**Issue by zhiyue** [zhiyue/mirror-blog#9]
machine learning

- Tutorial
  -  Deep Learning Tutorial: Perceptrons to Machine Learning Algorithms | Toptal 
""A Deep Learning Tutorial: From Perceptrons to Deep Networks"
"
  -  A Machine Learning Introductory Tutorial with Examples | Toptal 
  -  Johnhacker/johnhacker.github.io: ai相关网址导航 
    -  搬运了一个 AI 和机器学习相关的资源，做了个网址导航，欢迎提交中文资源 - V2EX 
  -  Deep Learning Glossary – WildML 
  -  Dive into Machine Learning by hangtwenty 
    - "Tools you'll need"
    - "Immerse yourself"
    - "Recommended course"
    - "Cheat sheets"
    - "Questions, answers, chats"
    - "Assorted Opinions and Other Resources"
  -  Machine Learning for Developers by Mike de Waard 
  -  Spandan-Madan/DeepLearningProject 
- Books
  -  Deep Learning 
    -  HFTrader/DeepLearningBook: MIT Deep Learning Book in PDF format 
    -  exacity/deeplearningbook-chinese: Deep Learning Book Chinese Translation 
  -  Neural networks and deep learning 
  - 
- Courses
  - cs231N
  - cs224d
  -  CS 294 Deep Reinforcement Learning, Spring 2017 
  -  UvA Deep Learning Course 
  -  CS281: Advanced Machine Learning 
  -  CSE 705: Deep Learning 
  -  CS584: Deep Learning 
- Videos and Lectures
  -  oxford-cs-deepnlp-2017/lectures: Oxford Deep NLP 2017 course 
    -  oxford-cs-deepnlp-2017/practical1: Oxford Deep NLP 2017 course - Practical 1: word2vec 
    -  oxford-cs-deepnlp-2017/practical2: Oxford Deep NLP 2017 course - Practical 2: Text Classification 
    -  oxford-cs-deepnlp-2017/practical3: Oxford Deep NLP 2017 course - Practical 3: Text Classification with RNNs 
- 
- Notes
  -  ahangchen/GDLnotes: Google Deep Learning Notes（TensorFlow教程） 
- 
- sites
  - 英文
    -  WildML – AI, Deep Learning, NLP 
    -  Sebastian Ruder 
    - 
  - 中文
    -  我爱机器学习|机器学习干货站 
    -  科学空间|Scientific Spaces 
    -  机器之心_专业的人工智能媒体和产业服务平台 
    -  PaperWeekly 
    -  主页 · ZMonster's Blog 
    -  SharEDITor - 关注大数据技术 
    - 
- papers
- subscribe
  -  The Wild Week in AI | Revue 
- FQA
  -  深度学习如何入门？ - 知乎 
    - "神经网络肯定是要学习的，主要是BP算法，可以看看PRML3、4、5三章"
    - "有一本系统的介绍深度学习的书，不过还没写完，样稿见DEEP LEARNING"
    - "Deep Learning Tutorial from Stanford - Stanford计算机系的官方tutorial，Andrew Ng执笔。要想了解DL的原理，这个最好用了。"
  -  What is the difference between Data Analytics, Data Analysis, Data Mining, Data Science, Machine Learning, and Big Data? - Quora 
  -  Machine Learning FAQ 
- sites
  -  The Wild Week in AI | Revue 
- Datasets
  - MNIST Handwritten digits
  - Google House Numbers from street view
  - CIFAR-10 and CIFAR-100
  - IMAGENET
  - Tiny Images 80 Million tiny images6.
  - Flickr Data 100 Million Yahoo dataset
  - Berkeley Segmentation Dataset 500
  - UC Irvine Machine Learning Repository
  - Flickr 8k
  - Flickr 30k
  - Microsoft COCO
  - VQA
  - Image QA
  - AT&T Laboratories Cambridge face database
  - AVHRR Pathfinder
  - Air Freight - The Air Freight data set is a ray-traced image sequence along with ground truth segmentation based on textural characteristics. (455 images + GT, each 160x120 pixels). (Formats: PNG)
  - Amsterdam Library of Object Images - ALOI is a color image collection of one-thousand small objects, recorded for scientific purposes. In order to capture the sensory variation in object recordings, we systematically varied viewing angle, illumination angle, and illumination color for each object, and additionally captured wide-baseline stereo images. We recorded over a hundred images of each object, yielding a total of 110,250 images for the collection. (Formats: png)
  - Annotated face, hand, cardiac & meat images - Most images & annotations are supplemented by various ASM/AAM analyses using the AAM-API. (Formats: bmp,asf)
  - Image Analysis and Computer Graphics
  - Brown University Stimuli - A variety of datasets including geons, objects, and "greebles". Good for testing recognition algorithms. (Formats: pict)
  - CAVIAR video sequences of mall and public space behavior - 90K video frames in 90 sequences of various human activities, with XML ground truth of detection and behavior classification (Formats: MPEG2 & JPEG)
  - Machine Vision Unit
  - CCITT Fax standard images - 8 images (Formats: gif)
  - CMU CIL's Stereo Data with Ground Truth - 3 sets of 11 images, including color tiff images with spectroradiometry (Formats: gif, tiff)
  - CMU PIE Database - A database of 41,368 face images of 68 people captured under 13 poses, 43 illuminations conditions, and with 4 different expressions.
  - CMU VASC Image Database - Images, sequences, stereo pairs (thousands of images) (Formats: Sun Rasterimage)
  - Caltech Image Database - about 20 images - mostly top-down views of small objects and toys. (Formats: GIF)
  - Columbia-Utrecht Reflectance and Texture Database - Texture and reflectance measurements for over 60 samples of 3D texture, observed with over 200 different combinations of viewing and illumination directions. (Formats: bmp)
  - Computational Colour Constancy Data - A dataset oriented towards computational color constancy, but useful for computer vision in general. It includes synthetic data, camera sensor data, and over 700 images. (Formats: tiff)
  - Computational Vision Lab
  - Content-based image retrieval database - 11 sets of color images for testing algorithms for content-based retrieval. Most sets have a description file with names of objects in each image. (Formats: jpg)
  - Efficient Content-based Retrieval Group
  - Densely Sampled View Spheres - Densely sampled view spheres - upper half of the view sphere of two toy objects with 2500 images each. (Formats: tiff)
  - Computer Science VII (Graphical Systems)
  - Digital Embryos - Digital embryos are novel objects which may be used to develop and test object recognition systems. They have an organic appearance. (Formats: various formats are available on request)
  - Univerity of Minnesota Vision Lab
  - El Salvador Atlas of Gastrointestinal VideoEndoscopy - Images and Videos of his-res of studies taken from Gastrointestinal Video endoscopy. (Formats: jpg, mpg, gif)
  - FG-NET Facial Aging Database - Database contains 1002 face images showing subjects at different ages. (Formats: jpg)
  - FVC2000 Fingerprint Databases - FVC2000 is the First International Competition for Fingerprint Verification Algorithms. Four fingerprint databases constitute the FVC2000 benchmark (3520 fingerprints in all).
  - Biometric Systems Lab - University of Bologna
  - Face and Gesture images and image sequences - Several image datasets of faces and gestures that are ground truth annotated for benchmarking
  - German Fingerspelling Database - The database contains 35 gestures and consists of 1400 image sequences that contain gestures of 20 different persons recorded under non-uniform daylight lighting conditions. (Formats: mpg,jpg)
  - Language Processing and Pattern Recognition
  - Groningen Natural Image Database - 4000+ 1536x1024 (16 bit) calibrated outdoor images (Formats: homebrew)
  - ICG Testhouse sequence - 2 turntable sequences from ifferent viewing heights, 36 images each, resolution 1000x750, color (Formats: PPM)
  - Institute of Computer Graphics and Vision
  - IEN Image Library - 1000+ images, mostly outdoor sequences (Formats: raw, ppm)
  - INRIA's Syntim images database - 15 color image of simple objects (Formats: gif)
  - INRIA
  - INRIA's Syntim stereo databases - 34 calibrated color stereo pairs (Formats: gif)
  - Image Analysis Laboratory - Images obtained from a variety of imaging modalities -- raw CFA images, range images and a host of "medical images". (Formats: homebrew)
  - NLM HyperDoc Visible Human Project - Color, CAT and MRI image samples - over 30 images (Formats: jpeg)
  - Middlebury Stereo Data Sets with Ground Truth - Six multi-frame stereo data sets of scenes containing planar regions. Each data set contains 9 color images and subpixel-accuracy ground-truth data. (Formats: ppm)
  - Computer Vision Group
  - Sequences for Flow Based Reconstruction - synthetic sequence for testing structure from motion algorithms (Formats: pgm)
  - Stereo Images with Ground Truth Disparity and Occlusion - a small set of synthetic images of a hallway with varying amounts of noise added. Use these images to benchmark your stereo algorithm. (Formats: raw, viff (khoros), or tiff)
  - Photometric 3D Surface Texture Database - This is the first 3D texture database which provides both full real surface rotations and registered photometric stereo data (30 textures, 1680 images). (Formats: TIFF)
  - Department Image Understanding
  - The AR Face Database - Contains over 4,000 color images corresponding to 126 people's faces (70 men and 56 women). Frontal views with variations in facial expressions, illumination, and occlusions. (Formats: RAW (RGB 24-bit))
  - Purdue Robot Vision Lab
  - The MIT-CSAIL Database of Objects and Scenes - Database for testing multiclass object detection and scene recognition algorithms. Over 72,000 images with 2873 annotated frames. More than 50 annotated object classes. (Formats: jpg)
  - The RVL SPEC-DB (SPECularity DataBase) - A collection of over 300 real images of 100 objects taken under three different illuminaiton conditions (Diffuse/Ambient/Directed). -- Use these images to test algorithms for detecting and compensating specular highlights in color images. (Formats: TIFF )
  - LIMSI-CNRS
  - Robot Vision Laboratory
  - LIMSI-CNRS/CHM/IMM/vision
  - The Xm2vts database - The XM2VTSDB contains four digital recordings of 295 people taken over a period of four months. This database contains both image and video data of faces.
  - Modis Airborne simulator, Gallery and data set - High Altitude Imagery from around the world for environmental modeling in support of NASA EOS program (Formats: JPG and HDF)
  - Centre for Vision, Speech and Signal Processing
  - Vision Research Group
  - Traffic Image Sequences and 'Marbled Block' Sequence - thousands of frames of digitized traffic image sequences as well as the 'Marbled Block' sequence (grayscale images) (Formats: GIF)
  - Otago Optical Flow Evaluation Sequences - Synthetic and real sequences with machine-readable ground truth optical flow fields, plus tools to generate ground truth for new sequences. (Formats: ppm,tif,homebrew)
  - IAKS/KOGS
  - Signal Analysis and Machine Perception Laboratory
  - U Bern Face images - hundreds of images (Formats: Sun rasterfile)
  - NIST Fingerprint data - compressed multipart uuencoded tar file
  - U Michigan textures (Formats: compressed raw)
  - OSU (MSU/WSU) Range Image Database - Hundreds of real and synthetic images (Formats: gif, homebrew)
  - U Oulu wood and knots database - Includes classifications - 1000+ color images (Formats: ppm)
  - OSU (MSU) 3D Object Model Database - several sets of 3D object models collected over several years to use in object recognition research (Formats: homebrew, vrml)
  - UCID - an Uncompressed Colour Image Database - a benchmark database for image retrieval with predefined ground truth. (Formats: tiff)
  - Geometric & Intelligent Computing Laboratory
  - UMass Vision Image Archive - Large image database with aerial, space, stereo, medical images and more. (Formats: homebrew)
  - Stuttgart Range Image Database - A collection of synthetic range images taken from high-resolution polygonal models available on the web (Formats: homebrew)
  - UNC's 3D image database - many images (Formats: GIF)
  - Mammography Image Databases - 100 or more images of mammograms with ground truth. Additional images available by request, and links to several other mammography databases are provided. (Formats: homebrew)
  - Image Database - An image database including some textures
  - USF Range Image Data with Segmentation Ground Truth - 80 image sets (Formats: Sun rasterimage)
  - University of Oulu Physics-based Face Database - contains color images of faces under different illuminants and camera calibration conditions as well as skin spectral reflectance measurements of each person.
  - Image Analysis Laboratory
  - Machine Vision and Media Processing Unit
  - MIT Vision Texture - Image archive (100+ images) (Formats: ppm)
  - University of Oulu Texture Database - Database of 320 surface textures, each captured under three illuminants, six spatial resolutions and nine rotation angles. A set of test suites is also provided so that texture segmentation, classification, and retrieval algorithms can be tested in a standard manner. (Formats: bmp, ras, xv)
  - JISCT Stereo Evaluation - 44 image pairs. These data have been used in an evaluation of stereo analysis, as described in the April 1993 ARPA Image Understanding Workshop paper ``The JISCT Stereo Evaluation'' by R.C.Bolles, H.H.Baker, and M.J.Hannah, 263--274 (Formats: SSI)
  - Machine Vision Group
  - ATR Research, Kyoto, Japan
  - Usenix face database - Thousands of face images from many different sites (circa 994)
  - JAFFE Facial Expression Image Database - The JAFFE database consists of 213 images of Japanese female subjects posing 6 basic facial expressions as well as a neutral pose. Ratings on emotion adjectives are also available, free of charge, for research purposes. (Formats: TIFF Grayscale images.)
  - View Sphere Database - Images of 8 objects seen from many different view points. The view sphere is sampled using a geodesic with 172 images/sphere. Two sets for training and testing are available. (Formats: ppm)
  - SEQUENCES FOR OPTICAL FLOW ANALYSIS (SOFA) - 9 synthetic sequences designed for testing motion analysis applications, including full ground truth of motion and camera parameters. (Formats: gif)
  - PRIMA, GRAVIR
  - ftp://ftp.limsi.fr/pub/quenot/opflow/testdata/piv/ - Real and synthetic image sequences used for testing a Particle Image Velocimetry application. These images may be used for the test of optical flow and image matching algorithms. (Formats: pgm (raw))
  - Vision-list Imagery Archive - Many images, many formats
  - OSU/SAMPL Database: Range Images, 3D Models, Stills, Motion Sequences - Over 1000 range images, 3D object models, still images and motion sequences (Formats: gif, ppm, vrml, homebrew)
  - Wiry Object Recognition Database - Thousands of images of a cart, ladder, stool, bicycle, chairs, and cluttered scenes with ground truth labelings of edges and regions. (Formats: jpg)
  - National Design Repository - Over 55,000 3D CAD and solid models of (mostly) mechanical/machined engineerign designs. (Formats: gif,vrml,wrl,stp,sat)
  - 3D Vision Group
  - NIST Fingerprint and handwriting - datasets - thousands of images (Formats: unknown)
  - Yale Face Database - 165 images (15 individuals) with different lighting, expression, and occlusion configurations.
  - Middlebury Stereo Vision Research Page - Middlebury College
  - Yale Face Database B - 5760 single light source images of 10 subjects each seen under 576 viewing conditions (9 poses x 64 illumination conditions). (Formats: PGM)
  - ftp://ftp.cps.msu.edu/pub/prip - many images (Formats: unknown)
  - Center for Computational Vision and Control
  - Machine Vision - Images from the textbook by Jain, Kasturi, Schunck (20+ images) (Formats: GIF TIFF)
  - DeepMind QA Corpus - Textual QA corpus from CNN and DailyMail. More than 300K documents in total. Paper for reference.
  - MIT face images and more - hundreds of images (Formats: homebrew)
- Podcasts
  - Linear Digressions
  - Data Skeptic
  - Talking Machines
  - The Data Show
  - Programming Throwdown [not strictly machine learning but very informative about programming]
- competitions
- awesome lists
  -  InnoArchiTech/datascience-ai-machinelearning-resources 
  -  Materials for Learning Machine Learning | Jack Simpson 
  -  awesome: Curated list of awesome lists 
  -  dive-into-machine-learning: Dive into Machine Learning with Python Jupyter notebook and scikit-learn 
  -  Machine Learning with Python - Google 文档 
  -  awesome-deep-learning: A curated list of awesome Deep Learning tutorials, projects and communities. 
  -  机器学习(Machine Learning)&深度学习(Deep Learning)资料(Chapter 1) 
  -  机器学习(Machine Learning)&深度学习(Deep Learning)资料(Chapter 2) 
  -  josephmisiti/awesome-machine-learning: A curated list of awesome Machine Learning frameworks, libraries and software. 
  -  ChristosChristofidis/awesome-deep-learning: A curated list of awesome Deep Learning tutorials, projects and communities. 
  -  awesome/machine-learning.md at master · shenwei356/awesome 
  -  ty4z2008/Qix: Machine Learning、Deep Learning、PostgreSQL、Distributed System、Node.Js、Golang 
  -  szwed/awesome-machine-learning-python: Machine and Deep Learning in Python 
  -  jobbole/awesome-machine-learning-cn: 机器学习资源大全中文版，包括机器学习领域的框架、库以及软件 
  -  songrotek/Deep-Learning-Papers-Reading-Roadmap: Deep Learning papers reading roadmap for anyone who are eager to learn this amazing tech! 
  -  terryum/awesome-deep-learning-papers: The most cited deep learning papers 
  -  Deep Learning | Jeremy D. Jackson, PhD 
  -  Deep Learning Courses - Machine Learning Mastery 
- example
  - iPython Notebooks
    -  Data-Analysis-and-Machine-Learning-Projects: Repository of teaching materials, code, and data for my data analysis and machine learning projects. 
      -  Example Machine Learning Notebook.ipynb 
- Frameworks
  - TensorFlow
    -  tensorflow/tensorflow: Computation using data flow graphs for scalable machine learning 
    -  vahidk/EffectiveTensorflow 
    -  一点关于Tensorflow学习的资料 – Snake 
- repos
- posts
-  Neural networks and deep learning 
-  josephmisiti/awesome-machine-learning: A curated list of awesome Machine Learning frameworks, libraries and software. 
-  index.md 
-  spro/practical-pytorch: Project based PyTorch tutorials, focused on using neural networks (especially RNNs) for natural language tasks 
-  【不周山之机器学习】贰 深度学习入门指南（基于 TensorFlow） | 小土刀 
-  Data Science - Page 1 - The Marketing Technologist | The marketing technology blog 
-  CSE 705: Deep Learning 
-  metrofun/machine-learning-surveys: A curated list of Machine Learning Surveys, Tutorials and Books. 

Indent | Tab
-- | --
Outdent | Shift + Tab
Move bullet point up | Ctrl + Shift + Up
Move bullet point down | Ctrl + Shift + Down
Zoom in | Ctrl + Shift + Right
Zoom out | Ctrl + Shift + Left
Delete bullet point | Ctrl + Shift + Backspace
Duplicate bullet point | Ctrl + Shift + D
Add a note | Shift + Enter
Bold | Ctrl + B
Italic | Ctrl + I
strikethrough | Alt + Shift + 5
Link text | Ctrl + K
Open webpage in new tab | Ctrl + Click
